core/trino-main/src/main/java/io/trino/operator/OperatorFactories.java

```diff
@@ -28,11 +30,58 @@
     OperatorFactory innerJoin(
```

Comment:
This became very reperitive. Could we extract something like
```
static class JoinOperatorType {
  private final JoinNode.Type type;
  private final boolean outputSingleMatch;
  private final boolean waitForBuild;
  
  public static innerJoin(boolean outputSingleMatch, boolean waitForBuild) {..}
  public static probeOuterJoin(boolean outputSingleMatch) {...}
  public static lookupOuterJoin(boolean waitForBuild) {...}
  public static fullOuterJoin() {...}

  private JoinOperatorType(...);
}
```

then you could just have:
```
OperatorFactory join(
            int operatorId,
            PlanNodeId planNodeId,
            JoinOperatorType type,
            JoinBridgeManager<? extends PartitionedLookupSourceFactory> lookupSourceFactory,
            boolean hasFilter,
            List<Type> probeTypes,
            List<Integer> probeJoinChannel,
            OptionalInt probeHashChannel,
            Optional<List<Integer>> probeOutputChannels,
            BlockTypeOperators blockTypeOperators);

OperatorFactory spillingJoin(
            int operatorId,
            PlanNodeId planNodeId,
            JoinOperatorType type,
            JoinBridgeManager<? extends LookupSourceFactory> lookupSourceFactory
            boolean hasFilter,
            List<Type> probeTypes,
            List<Integer> probeJoinChannel,
            OptionalInt probeHashChannel,
            Optional<List<Integer>> probeOutputChannels,
            BlockTypeOperators blockTypeOperators);
```